packages/scenario/test/select.test.ts

```diff
@@ -0,0 +1,137 @@
+ */
+interface GetSelectFromRepeatFormOptions {
+	readonly substituteAbsoluteBodyReferences: boolean;
+}
+
+// Ported as of https://github.com/getodk/javarosa/commit/5ae68946c47419b83e7d28290132d846e457eea6
+describe.each<GetSelectFromRepeatFormOptions>([
+	{ substituteAbsoluteBodyReferences: false },
+	{ substituteAbsoluteBodyReferences: true },
+])(
+	'Select - JavaRosa ports (substituting absolute body references: $substituteAbsoluteBodyReferences)',
+	({ substituteAbsoluteBodyReferences }) => {
+		// // TODO: didn't need this for the first test, but it's here in case there
+		// // are subsequent tests which would use it. REMOVE THIS if it isn't used in
+		// // any tests submitted in the bulk test port PR.
+		// const relativeBodyRefTest = {
+		// 	/**
+		// 	 * Use for tests which fail **because** the form fixture uses relative
+		// 	 * body references.
+		// 	 */
+		// 	it: substituteAbsoluteBodyReferences ? it : it.fails,
+		// } as const;
+
+		const getSelectFromRepeatForm = (predicate = '') => {
+			const repeatValueInputRef = substituteAbsoluteBodyReferences ? '/data/repeat/value' : 'value';
+			const repeatLabelInputRef = substituteAbsoluteBodyReferences ? '/data/repeat/label' : 'label';
+			const filterInputRef = substituteAbsoluteBodyReferences ? '/data/filter' : 'filter';
+
+			return html(
+				head(
+					title('Select from repeat'),
+					model(
+						mainInstance(
+							t(
+								"data id='repeat-select'",
+								t('repeat', t('value'), t('label')),
+								t('filter'),
+								t('select')
+							)
+						)
+					)
+				),
+				body(
+					repeat('/data/repeat', input(repeatValueInputRef), input(repeatLabelInputRef)),
+					input(filterInputRef),
+					select1Dynamic('/data/select', '../repeat' + (predicate !== '' ? `[${predicate}]` : ''))
+				)
+			);
+		};
+
+		/**
+		 * Integration tests to verify that the choice lists for "dynamic selects"
+		 * (selects with itemsets rather than inline items) are updated when
+		 * dependent values change.
+		 *
+		 * See also:
+		 * - {@see SelectOneChoiceFilterTest}
+		 * - {@see SelectMultipleChoiceFilterTest} for coverage of dynamic select
+		 *   multiples
+		 * - {@see XPathFuncExprRandomizeTest} for coverage of choice list updates
+		 *   when randomization is specified
+		 *
+		 * **PORTING NOTES**
+		 *
+		 * 1. The above reference to `XPathFuncExprRandomizeTest` doesn't resolve to
+		 *    anything here, but it evidently doesn't resolve to anything in
+		 *    JavaRosa (anymore?) either.
+		 *
+		 * 2. Despite accommodating relative body `ref` attributes, this test still
+		 *    fails. A brief side quest to investigate the nature of the failure
+		 *    revealed that:
+		 *
+		 *    - Even without supporting relative `ref`s on controls, we'll need to
+		 *      do so for `<itemset>` and its `<value>` child (presumably its
+		 *      `<label>` child as well). The concern is so general we probably
+		 *      might as well actually just support them all.
```

Comment:
All current itemset support has been focused on secondary instances. In terms of interaction with relative references, I believe the primary concern would be whether it establishes the appropriate reactive subscriptions. Contextualizing relative references hadn't been a particular concern for itemsets up to this point, because _being focused on secondary instances_ there's nothing reactive to subscribe to (secondary instances being static).